pyro/poutine/trace.py

```diff
@@ -1,8 +1,19 @@
 from __future__ import absolute_import, division, print_function
 
 import collections
+import warnings
 
 import networkx
+import numpy as np
+
+
+def _warn_if_nan(name, variable):
```

Comment:
Is this `name` argument usually something that is easy to interpret?

Reply:
It is identical to the "site_name" in `pyro.sample("site_name", dist.normal, sigma, mu)` [sic] or `pyro.observe("site_name", dist.normal, mu, sigma, x)` [sic].

Reply:
ok. so it wouldn't tell you whether the problem is in the guide or the model, but that is handled by examining the stack traces as you describe above I guess?

Reply:
Yeah this won't distinguish guide-vs-model unless you examine the stack trace. It would take a lot more plumbing to add that to the warning message. But I do agree that it would help. We'll keep our eye out for better ways to add model-vs-guide info to warning messages.